### PR TITLE
EREGCSC-1921 -- Statute SPA/Table

### PR DESCRIPTION
**Pull request by PhilR8**
Resolves [EREGCSC-1921](https://jiraent.cms.gov/browse/EREGCSC-1921)

**Description**

Creates Statute Table that loads data from all SSA titles into a single table.

**This pull request changes:**

- Makes `/statutes` a Single Page App to support future enhancements
- Adds Header/Footer and Banner
- Pulls Statute data from API endpoint at `v3/statutes`
- Renders API response in a table that matches [Figma](https://www.figma.com/file/mXCFChrIR50gbFAvXV0kgB/statute-to-USC-conversion?node-id=60%3A618&mode=dev)
- Styles table to match mobile and desktop Figma designs, as well as implements behavior notes from Figma

**Steps to manually verify this change:**

1. Visit [Experimental Deployment and go to the `/statutes` page](https://371wnfkrul.execute-api.us-east-1.amazonaws.com/dev856/statutes/)
2. View page in desktop widths (greater than 1024px) and narrow widths (less than 1024px) and make sure table matches Figma 
3. Poke around table and make sure links work, things look good, etc.

### CODE REVIEW

File: solution/ui/e2e/cypress/fixtures/statutes.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        "section": "1104",
+        "title": 42,
+        "usc": "1304",
+        "act": "Social Security Act",
+        "name": "Reservation of power.",
+        "statute_title": 11,
+        "statute_title_roman": "XI",
+        "source_url": "https://www.govinfo.gov/content/pkg/COMPS-8763/uslm/COMPS-8763.xml"
+    }
```
Comment (thwalker6) [inline solution/ui/e2e/cypress/fixtures/statutes.json:41]: Missing trailing comma.

Comment (PhilR8) [reply solution/ui/e2e/cypress/fixtures/statutes.json:41]: JSON is a bit different than JS in that it does not allow trailing commas.  See third paragraph here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas

File: solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/StatuteTable.vue [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                        :cell-data="column"
+                        :statute="statute"
+                    />
+                </tr>
+            </tbody>
+        </table>
+    </div>
+</template>
+
+<style></style>
```
Comment (thwalker6) [inline solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/StatuteTable.vue:90]: is it necessary to have empty style tags here?  We should just leave those off no?

File: solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/utils/urlMethods.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+const houseGovUrl = ({ title, usc }) =>`https://uscode.house.gov/view.xhtml?hl=false&edition=prelim&req=granuleid%3AUSC-prelim-title${title}-section${usc}`;
+
+/**
+ * @param {string} souce_url - URL containing COMPS number
+ *
+ * @returns {string} url - url to GovInfo.gov page displaying Statute Compilation PDF
+ */
+const statuteCompilationUrl = ({ source_url }) => {
+    const compsNumber = source_url
+        .split("/")
```
Comment (thwalker6) [inline solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/utils/urlMethods.js:17]: Can you add a vitest for this utility function?

**Review by thwalker6 — CHANGES_REQUESTED** [review]
Just add the one test and I'm cool with it.  Also do we want the empty style tags on the vue component

File: solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/StatuteTable.vue (hunk 2 above)
Comment (PhilR8) [reply solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/StatuteTable.vue:90]: I'll remove it from this component, and going forward only include it if we use it.

File: solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/utils/urlMethods.js (hunk 3 above)
Comment (PhilR8) [reply solution/ui/regulations/eregs-component-lib/src/components/shared-components/Statutes/utils/urlMethods.js:17]: I will in the next story, EREGCSC-1923.  That's the integration/polish story that will ship the feature to users.  I've already started a branch with some updated Cypress tests, but now I'm at the point where this PR needs to be in `main` so I can continue.

**Review by thwalker6 — APPROVED** [review]
lgtm

**Pull request merged**